controllers/argocd/util.go

```diff
@@ -1058,7 +1058,7 @@ func (r *ReconcileArgoCD) setResourceWatches(bldr *builder.Builder, clusterResou
 	}
 
 	// Watch for changes to primary resource ArgoCD
-	bldr.For(&argoproj.ArgoCD{}, builder.WithPredicates(deleteSSOPred, deleteNotificationsPred))
+	bldr.For(&argoproj.ArgoCD{}, builder.WithPredicates(deleteSSOPred, deleteNotificationsPred, argoCDNamespaceManagementFilterPredicate()))
```

Comment:
With the change I have proposed to `nmMapper` above, I don't think `argoCDNamespaceManagementFilterPredicate()` is required here.